namechooser_3000.py

```diff
@@ -52,21 +52,27 @@ def spin_bar():
         sys.stdout.write('\b')
     finished = True
 
-def get_random_number():
+def get_random_number(num_names):
     """Fetch data from the ANU Quantum Random Numbers JSON API"""
     global numbers
     if numbers:
-        return numbers.pop()
+        rn = numbers.pop()
```

Comment:
The usual way of handling random number generation is to seed a pseudo random number generator with the content from an entropy source ("ANU Quantum Random Numbers" in our case). In that way you would never run out of random numbers :)

You could for instance use the API to get a random uint16 number and use that for seeding using random.seed() and then call random.randint() to get the correct number.

Reply:
Sounds like a good idea!